script.js

```diff
@@ -128,6 +130,7 @@ const fetchStockInfo = async (ticker_symbol) => {
 //Main function for the math
 const stockMathInfo = async (year_number, month_number, day_number, symbol) => {
 
+    // COMMENT(andymina)
   //Calling the information from the fetch call
   await fetchStockInfo(symbol)
```

Comment:
Splitting stockMathInfo would probably mean also passing the API data in as a parameter. Your new search button event handler might looking something like:

```javascript
search_btn.addEventListener("click", async (event) => {
	// all of your preprocessing here
	
	const stockData = await fetchStockInfo(symbol);
	const average = stockMathInfo(year, month, day, symbol, stockData);

	// post-processing
});
```

Splitting the two functions removes the need for stockMathInfo to be async-await and make it a lot easier to implement the look-forward approach because you can just append your predictions to `stockData` and repeatedly call that in a loop for each of the new days. This would probably look like:

```javascript
search_btn.addEventListener("click", async (event) => {
	// all of your preprocessing here
	
	let stockData = await fetchStockInfo(symbol);
	
	let  lookForwardWindow = 3;
	for (let i = 0; i < lookForwardWindow; i++) {
		let currentDay = day + i;
		let average = stockMathInfo(year, month, currentData, symbol, stockData);
		prices.push(average);
	}

	// post-processing
});
```